dvc/tree/base.py

```diff
@@ -46,8 +47,8 @@ def use_state(call):
 
 class BaseTree:
     scheme = "base"
-    REQUIRES = {}
-    PATH_CLS = URLInfo
+    REQUIRES: ClassVar[Dict[str, str]] = {}
+    PATH_CLS = URLInfo  # type: Any
```

Comment:
Hack, not sure how to fix this. I think the best thing to do is, force types on all classes that override the types. But, it still won't solve our problem with function signature violating LSP. Generics might solve, but the `mypy` does not work on all cases. :(